services/asymkey/commit.go

```diff
@@ -115,7 +122,9 @@ func ParseCommitWithSignatureCommitter(ctx context.Context, c *git.Commit, commi
 			}
 		}
 
-		committerEmailAddresses, _ := user_model.GetEmailAddresses(ctx, committer.ID)
+		committerEmailAddresses, _ := cache.GetWithContextCache(ctx, cacheUserEmailAddressKey, committer.ID, func() ([]*user_model.EmailAddress, error) {
+			return user_model.GetEmailAddresses(ctx, committer.ID)
+		})
```

Comment:
Maybe it could use generic type:


```
committerEmailAddresses, _ := cache.GetWithContextCacheFunc(ctx, cacheUserEmailAddressKey, committer.ID, user_model.GetEmailAddresses)
```